keras/src/trainers/trainer.py

```diff
@@ -25,6 +26,9 @@ def __init__(self):
         self.steps_per_execution = 1
         # Can be set by callbacks in on_train_begin
         self._initial_epoch = None
+        self._compute_loss_has_training_arg = (
+            len(inspect.signature(self.compute_loss).parameters) >= 5
```

Comment:
Can we look at the arg names, rather? That would be more robust

Reply:
Done.